examples/nucleo_f429zi/nanopb/main.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022, Petteri Aimonen <jpa at nanopb.mail.kapsi.fi>, Lucas Moesch
+ */
+
+#include <pb_encode.h>
+#include <pb_decode.h>
+#include "protocol/example.pb.hpp"
+
+int main()
+{
+    /* This is the buffer where we will store our message. */
+    uint8_t buffer[128];
+    size_t message_length;
+    bool status;
+
+    {
+        /* Allocate space on the stack to store the message data.
+         *
+         * Nanopb generates simple struct definitions for all the messages.
+         * - check out the contents of simple.pb.h!
+         * It is a good idea to always initialize your structures
+         * so that you do not have garbage data from RAM in there.
+         */
+        SimpleMessage message = SimpleMessage_init_zero;
+
+        /* Create a stream that will write to our buffer. */
+        pb_ostream_t stream = pb_ostream_from_buffer(buffer, sizeof(buffer));
+
+        /* Fill in the lucky number */
+        message.lucky_number = 13;
+
+        /* Now we are ready to encode the message! */
+        status = pb_encode(&stream, SimpleMessage_fields, &message);
+        message_length = stream.bytes_written;
+
+        /* Then just check for any errors.. */
+        if (!status)
+        {
+            return 1;
```

Comment:
Ok, but FYI this will reboot the device eventually.

Reply:
Right, I just copy-pasted the nanopb example, hence the different copyright. I will adjust it so it runs on the uC.